src/scripts/main.js

```diff
@@ -1 +1,80 @@
 'use strict';
+
+const promise1 = new Promise((resolve, reject) => {
+  document.addEventListener('click', () => {
+    resolve('First promise was resolved');
+  });
+
+  setTimeout(reject, 3000);
+});
+
+promise1
+  .then(resolveHandler)
+  .catch(rejectHandler);
+
+const promise2 = new Promise((resolve, reject) => {
+  document.addEventListener('click', () => {
+    resolve('Second promise was resolved');
+  });
+
+  document.addEventListener('contextmenu', () => {
+    event.preventDefault();
+    resolve('Second promise was resolved');
+  });
+});
+
+promise2
+  .then(resolveHandler)
+  .catch(rejectHandler);
+
+const promise3 = new Promise((resolve, reject) => {
+  let leftClick = false;
+  let rightClick = false;
+
+  document.addEventListener('click', () => {
+    leftClick = true;
+
+    if (leftClick && rightClick) {
+      resolve(`Third promise was resolved`);
+    }
+  });
+
+  document.addEventListener('contextmenu', () => {
+    rightClick = true;
+
+    if (leftClick && rightClick) {
+      event.preventDefault();
+      resolve(`Third promise was resolved`);
+    }
+  });
+});
+
+promise3
+  .then(resolveHandler)
+  .catch(rejectHandler);
+
+function resolveHandler(result) {
```

Comment:
```suggestion
function handleResolve(result) {
```
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364